Don't hardcode -fpic, some of our architectures need -fPIC.
Don't hardcode -O3.

Index: src/build/Common.gmake
--- src/build/Common.gmake.orig
+++ src/build/Common.gmake
@@ -279,7 +279,6 @@ ifeq ($(machine_uname),x86_64)
  machine_uname	:= x64
 endif
 ifndef win
- CFLAGS +=	-fpic
 endif
 ifeq ($(machine_uname),sparc64)
  CFLAGS +=      -D__BIG_ENDIAN__
@@ -524,7 +523,7 @@ ifdef DEBUG
 else # RELEASE
  # -finline-functions (used to) break the baja build badly.
  # This also meant that -O3 wouldn't work either.
- CFLAGS	:= -O3 -Wno-unused-result -ffast-math -funroll-loops $(CFLAGS) -DNDEBUG
+ CFLAGS	:= -Wno-unused-result -ffast-math -funroll-loops $(CFLAGS) -DNDEBUG
  ifndef NO_LTO
   # Not Yet... this requires more care with library ordering.
   # CFLAGS += -flto
